scripts/makeheader

```diff
@@ -3,9 +3,41 @@ from __future__ import print_function
 import re
 import os
 
-# expected format:
-# #define XF86XK_FooBar _EVDEVK(0x123) /* some optional comment */
-evdev_pattern = re.compile(r'^#define\s+XF86XK_(?P<name>\w+)\s+_EVDEVK\((?P<value>0x[0-9A-Fa-f]+)\)')
+# Expected format:
+#     #define XF86XK_FooBar 0x1234         /* some optional comment */
+# or:
+#     #define XF86XK_FooBar _EVDEVK(0x123) /* some optional comment */
+# We also need to match commented evdev entries:
+#     /* Use: XF86XK_FooBar _EVDEVK(0x123)    some optional comment */
+keysym_entry_pattern = re.compile(
+    r'''^
+    (?P<define>\#define|/\*\s+Use:)\s+
+    (?P<prefix>\w*)XK_(?P<name>\w+)(?P<spacing>\s+)
+    (?P<evdev>_EVDEVK\()?(?P<value>0x[0-9A-Fa-f]+)(?(evdev)\))
+    ''',
+    re.VERBOSE,
+)
+# Match remaining XK_ references
+xorgproto_keysym_prefix_pattern = re.compile(r'\b(?P<prefix>\w*)XK_(?!KOREAN\b)')
```

Comment:
got confused by this and had to check the various header files, would be nice to add something like:
```
# XK_KOREAN has ifdefs, ensure we handle those too
```

Reply:
It is confusing, indeed! This pattern is intended to match remaining keysyms references in _comments_, e.g. replace:

```
#define XKB_KEY_SunCompose		0x0000FF20	/* Same as XK_Multi_key */
```

with:

```
#define XKB_KEY_SunCompose		0x0000FF20	/* Same as XKB_KEY_Multi_key */
```

I updated the comment.